File: JS/Programmers/Level1/07 로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
  var answer = [];

  var num = 0;
  var zero = 0;

  for (var i = 0; i < lottos.length; i++) {
    if (lottos[i] == 0) {
      zero++;
    } else if (win_nums.includes(lottos[i]) == true) {
      num++;
    }
  }

  switch (zero + num) {
    case 1:
      answer.push(6);
      break;
    case 2:
      answer.push(5);
      break;
    case 3:
      answer.push(4);
      break;
    case 4:
      answer.push(3);
      break;
    case 5:
      answer.push(2);
      break;
    case 6:
      answer.push(1);
      break;
    default:
      answer.push(6);
      break;
  }

  switch (num) {
    case 1:
      answer.push(6);
      break;
    case 2:
      answer.push(5);
      break;
    case 3:
      answer.push(4);
      break;
    case 4:
      answer.push(3);
      break;
    case 5:
      answer.push(2);
      break;
    case 6:
      answer.push(1);
      break;
    default:
      answer.push(6);
      break;
  }

  return answer;
}

File: C++/Programmers/Level1/07 로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;

    int num = 0;
    int zero = 0;

    for(int i=0; i<lottos.size(); i++){
        if(lottos[i] == 0){
            zero++;
        }
        else if(find(win_nums.begin(), win_nums.end(), lottos[i])!=win_nums.end())
            num++;
    }

    switch (zero+num)
    {
    case 1: answer.push_back(6);
        break;
    case 2: answer.push_back(5);
        break;
    case 3: answer.push_back(4);
        break;
    case 4: answer.push_back(3);
        break;;
    case 5: answer.push_back(2);
        break;
    case 6: answer.push_back(1);
        break;
    default:
        answer.push_back(6);
        break;
    }

    switch(num){
    case 1: answer.push_back(6);
        break;
    case 2: answer.push_back(5);
        break;
    case 3: answer.push_back(4);
        break;
    case 4: answer.push_back(3);
        break;;
    case 5: answer.push_back(2);
        break;
    case 6: answer.push_back(1);
        break;
    default:
        answer.push_back(6);
        break;
    }


    return answer;
}